toffy/streak_detection.py

```diff
@@ -0,0 +1,410 @@
+        the intensity. Defaults to 95.
+
+    Returns:
+        np.ndarray: The binary mask containing all of the candidate strokes.
+    """
+    # Denoise the Image
+    input_image = restoration.denoise_wavelet(
+        input_image, wavelet="db2", mode="hard", rescale_sigma=True
+    )
+    # Rescale the intensity using percentile ranges
+    pmin_v, pmax_v = np.percentile(input_image, (pmin, pmax))
+    input_image = exposure.rescale_intensity(input_image, in_range=(pmin_v, pmax_v))
+    # Laplace filter to get the streaks
+    input_image = filters.laplace(input_image, ksize=3)
+    input_image = exposure.rescale_intensity(input_image, out_range=(0, 1))
+    # Smoothing
+    input_image = filters.gaussian(input_image, sigma=(0, gaussian_sigma))  # (y, x)
+    # Exposure Adjustments
+    input_image = exposure.adjust_gamma(input_image, gamma=gamma, gain=gamma_gain)
+    input_image = exposure.adjust_log(input_image, gain=log_gain, inv=True)
+    input_image = exposure.rescale_intensity(input_image, out_range=(0, 1))
+    # apply threshold
+    binary_mask = input_image > 0.3
+
+    return binary_mask
+
+
+def _filter_mask(streak_data: StreakData, min_length: int = 50) -> None:
+    """Filters the streaks. Removes incorrectly masked streaks which are too short and do not have
+    a high eccentricity.
+
+    Args:
+        streak_data (StreakData): An instance of the StreakData Dataclass, holds all necessary
+        data for streak correction.
+        min_length (int): The lower threshold for filtering streaks in pixels. Defaults to 50.
+    """
+    # Label all the candidate streaks
+    labeled_streaks = measure.label(
+        streak_data.streak_mask, connectivity=2, return_num=False
+    )
+
+    # Gather properties of all the candidate streaks using regionprops.
+    region_properties = measure.regionprops_table(
+        label_image=labeled_streaks,
+        cache=True,
+        properties=[
+            "label",
+            "bbox",
+            "eccentricity",
+            "area",
+        ],
+    )
+
+    # Convert dictionary of region properties to DataFrame.
+    streak_data.streak_df = pd.DataFrame(region_properties)
+
+    # Rename the bounding box columns.
+    streak_data.streak_df.rename(
+        {
+            "bbox-0": "min_row",
+            "bbox-1": "min_col",
+            "bbox-2": "max_row",
+            "bbox-3": "max_col",
+        },
+        axis="columns",
+        inplace=True,
+    )
+    # Give the index column a name.
+    streak_data.streak_df.index.names = ["index"]
+
+    # Filter out eccentricities that are less than 0.99999 (only keep straight lines)
+    # Filter out small areas (small lines)
+    eccentricity_value = 0.99999
+    area_value = min_length
+    streak_data.filtered_streak_df = streak_data.streak_df.query(
+        f"eccentricity > {eccentricity_value} and area > {area_value}"
+    )
+
+
+def _make_filtered_mask(streak_data: StreakData) -> None:
```

Comment:
`_filter_mask` and `_make_filtered_mask` sound like the same function. The docstrings also don't clearly indicate what the differences are. 

The key piece of information missing from the function name and docstring is that `_filter_mask` is making a filtered df, not a filtered mask.